File: app/src/main/java/com/example/jacob/android_sprint1_challenge/EditActivity.java
package com.example.jacob.android_sprint1_challenge;

import android.app.Activity;
import android.arch.lifecycle.ViewModelProviders;
import android.content.Context;
import android.content.Intent;
import android.graphics.Paint;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Switch;

public class EditActivity extends AppCompatActivity {

    public static final String EDIT_MOVIE_KEY = "edit_movie";
    private Context context;
    EditText editTitle;
    Switch switchWatched;
    Movie movie;
    private MovieViewModel viewModel;

    public static final int DELETE_REQUEST_CODE = 2;

    public static final int SAVE_REQUEST_CODE = 3;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit);

        context = this;
        editTitle = findViewById(R.id.edit_title);
        switchWatched = findViewById(R.id.switch_watched);
        viewModel = ViewModelProviders.of(this).get(MovieViewModel.class);
        editTitle.requestFocus();

        movie = (Movie) getIntent().getSerializableExtra(EDIT_MOVIE_KEY);
        if (movie == null) {
            movie = new Movie(Movie.NO_ID);
            movie.setTitle("");
            movie.setWatched(false);
        } else {
            editTitle.setText(movie.getTitle());
            switchWatched.setChecked(movie.getWatched());
        }

        findViewById(R.id.button_save).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                movie.setTitle(editTitle.getText().toString());
                movie.setWatched(switchWatched.isChecked());
                Intent resultIntent = new Intent();
                resultIntent.putExtra(EDIT_MOVIE_KEY, movie);
                setResult(Activity.RESULT_OK, resultIntent);
                finish();
            }
        });

        findViewById(R.id.button_delete).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                viewModel.deleteMovie(movie);
                Intent resultIntent = new Intent();
                resultIntent.putExtra(EDIT_MOVIE_KEY, movie);
                setResult(Constants.DELETE_RESULT_CODE, resultIntent);
                finish();
            }
        });
    }


    @Override
    public void onBackPressed() {
        super.onBackPressed();
    }


    private void prepResult() {

    }
}

File: app/src/main/java/com/example/android_sprint1_challenge/ui/EditActivity.kt
package com.example.android_sprint1_challenge.ui

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.example.android_sprint1_challenge.R
import com.example.android_sprint1_challenge.model.MovieData
import kotlinx.android.synthetic.main.activity_edit.*
import kotlinx.android.synthetic.main.activity_main.*
import java.util.ArrayList

class EditActivity : AppCompatActivity() {



    companion object {
        const val MOVIE_KEY = "New Movie Key"

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit)


   var bundle:Bundle?=intent.extras
        if(bundle!=null) {
            var selectedMovie: MovieData = bundle!!.getSerializable("SelectedMovie") as MovieData
            editText_movieName.setText(selectedMovie.name)
            watched_switch.isChecked=selectedMovie.watched

        }


        button_save.setOnClickListener {
            saveMovie()
        }
        button_delete.setOnClickListener {

        }

    }

    fun saveMovie(){
        val resultIntent = Intent()
        resultIntent.putExtra(MOVIE_KEY,MovieData(editText_movieName.text.toString(),watched_switch.isChecked))
        setResult(Activity.RESULT_OK, resultIntent)
        finish() // destroy the activity
    }
    override fun onBackPressed() {
        saveMovie()
    }
}

